code/datums/repositories/crew.dm

```diff
@@ -23,76 +23,56 @@ var/global/datum/repository/crew/crew_repository = new()
+	for(var/t in tracked)
+		var/obj/item/clothing/under/C = t
 		var/turf/pos = get_turf(C)
 		if((C) && (C.has_sensor) && (pos) && (pos.z == z_level) && (C.sensor_mode != SUIT_SENSOR_OFF) && !within_jamming_range(C))
 			if(istype(C.loc, /mob/living/carbon/human))
 				var/mob/living/carbon/human/H = C.loc
 				if(H.w_uniform != C)
 					continue
 
-				var/list/crewmemberData = list("dead"=0, "area"="", "x"=-1, "y"=-1, "z"=-1, "level"="", "ref" = "\ref[H]")
+				var/list/crewmemberData = list("dead"=null, "area"=null, "x"=null, "y"=null, "z"=null, "level"=null, "ref" = "\ref[H]")
 
-				crewmemberData["sensor_type"] = C.sensor_mode
+				crewmemberData["stype"] = C.sensor_mode
 				crewmemberData["name"] = H.get_authentification_name(if_no_id="Unknown")
-				crewmemberData["rank"] = H.get_authentification_rank(if_no_id="Unknown", if_no_job="No Job")
-				crewmemberData["assignment"] = H.get_assignment(if_no_id="Unknown", if_no_job="No Job")
+				crewmemberData["ass"] = H.get_assignment(if_no_id="Unknown", if_no_job="No Job")
 
 				if(C.sensor_mode >= SUIT_SENSOR_BINARY)
 					crewmemberData["dead"] = H.stat > UNCONSCIOUS
 
 				if(C.sensor_mode >= SUIT_SENSOR_VITAL)
 					crewmemberData["pulse"] = "N/A"
-					crewmemberData["pulse_span"] = "neutral"
-					crewmemberData["true_pulse"] = -1
+					crewmemberData["tpulse"] = -1
 					if(!H.isSynthetic() && H.should_have_organ(BP_HEART))
 						var/obj/item/organ/internal/heart/O = H.internal_organs_by_name[BP_HEART]
 						if (!O || !BP_IS_ROBOTIC(O)) // Don't make medical freak out over prosthetic hearts
-							crewmemberData["true_pulse"] = H.pulse()
+							crewmemberData["tpulse"] = H.pulse()
 							crewmemberData["pulse"] = H.get_pulse(GETPULSE_TOOL)
-							switch(crewmemberData["true_pulse"])
-								if(PULSE_NONE)
-									crewmemberData["pulse_span"] = "bad"
-								if(PULSE_SLOW)
-									crewmemberData["pulse_span"] = "average"
-								if(PULSE_NORM)
-									crewmemberData["pulse_span"] = "good"
-								if(PULSE_FAST)
-									crewmemberData["pulse_span"] = "highlight"
-								if(PULSE_2FAST)
-									crewmemberData["pulse_span"] = "average"
-								if(PULSE_THREADY)
-									crewmemberData["pulse_span"] = "bad"
 
 					crewmemberData["pressure"] = "N/A"
-					crewmemberData["true_oxygenation"] = -1
-					crewmemberData["oxygenation"] = ""
-					crewmemberData["oxygenation_span"] = ""
+					crewmemberData["toxyg"] = -1
+					crewmemberData["oxyg"] = 0
 					if(!H.isSynthetic() && H.should_have_organ(BP_HEART))
 						crewmemberData["pressure"] = H.get_blood_pressure()
-						crewmemberData["true_oxygenation"] = H.get_blood_oxygenation()
-						switch (crewmemberData["true_oxygenation"])
+						crewmemberData["toxyg"] = H.get_blood_oxygenation()
+						switch (crewmemberData["toxyg"])
 							if(105 to INFINITY)
-								crewmemberData["oxygenation"] = "increased"
-								crewmemberData["oxygenation_span"] = "highlight"
+								crewmemberData["oxyg"] = 5
```

Comment:
Why is this a number?

Reply:
Because:
1. Numbers are cheap to compare.
2. Numbers take up less space.
3. UI only needs this number to understand what is the oxygenation state.

Reply:
String comparison in DM is pointer comparison. Ergo the same as integer comparison.

Use a define, magic numbers themselves are baaad.

Reply:
Agreed, define would be better